tests/FakeItEasy.Tests/TestHelpers/MethodDataAttribute.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace FakeItEasy.Tests.TestHelpers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Linq;
+    using System.Reflection;
+    using Xunit.Sdk;
+
+    /// <summary>
+    /// Provides a data source for a data theory, with the data coming from a public static method.
+    /// The target method must return the test data in something something compatible with IEnumerable&lt;object[]&gt;.
+    /// Caution: the method's result is completely enumerated before any test is run.
+    /// </summary>
+    [SuppressMessage("Microsoft.Design", "CA1019:DefineAccessorsForAttributeArguments",
+        Justification = "No need to access arguments at runtime.")]
+    [DataDiscoverer("Xunit.Sdk.MemberDataDiscoverer", "xunit.core")]
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public sealed class MethodDataAttribute : DataAttribute

```

Comment:
If I understand correctly, the difference between this and `MemberDataAttribute` is that it looks for the method on the `ReflectedType` rather than the `DeclaringType`? Maybe the name should reflect that (not sure how, though)